trace_decoder/src/lib.rs

```diff
@@ -109,30 +63,402 @@
 //! - if there are withdrawals, one dummy transaction is added at the end, with
 //!   all the withdrawals in it.
 
-#![feature(linked_list_cursors)]
-#![feature(trait_alias)]
-#![feature(iter_array_chunks)]
 #![deny(rustdoc::broken_intra_doc_links)]
-#![deny(missing_debug_implementations)]
-#![deny(missing_docs)]
-
-#[cfg(doc)]
-use {
-    trace_protocol::{BlockTrace, TxnInfo},
-    types::OtherBlockData,
-};
-
-/// Provides debugging tools and a compact representation of state and storage
-/// tries, used in tests.
-pub mod compact;
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+
+/// The broad overview is as follows:
+///
+/// 1. Ethereum nodes emit a bunch of binary [`wire::Instruction`]s, which are
+///    parsed in [`wire`].
+/// 2. They are passed to one of two "frontends", depending on the node
+///    - [`hermez_cdk_erigon`], which contains an [`smt_trie`].
+///    - [`zero_jerigon`], which contains an [`mpt_trie`].
+/// 3. The frontend ([`hermez_cdk_erigon::Frontend`] or
+///    [`zero_jerigon::Frontend`]) is passed to the "backend", which lowers to
+///    [`evm_arithmetization::GenerationInputs`].
+///
+/// Deviations from the specification are signalled with `BUG(spec)` in the
+/// code.
+const _DEVELOPER_DOCS: () = ();
```

Comment:
Do you still need this?

Reply:
normally developer documentation is `// <some docs>` but they run the risk of getting stale that way if they link to items in the crate. This is the way I do developer docs with intra-doc links. I _do_ think this developer documentation is valuable, so I want to keep it